src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,541 @@
+ * @param path              pointer to a path
+ */
+void lv_vector_path_delete(lv_vector_path_t * path);
+
+/**
+ * Begin a new sub path and set a point to path
+ * @param path              pointer to a path
+ * @param p                 pointer to a `lv_fpoint_t` variable
+ */
+void lv_vector_path_move_to(lv_vector_path_t * path, const lv_fpoint_t * p);
+
+/**
+ * Add a line to the path from last point to the point
+ * @param path              pointer to a path
+ * @param p                 pointer to a `lv_fpoint_t` variable
+ */
+void lv_vector_path_line_to(lv_vector_path_t * path, const lv_fpoint_t * p);
+
+/**
+ * Add a quadratic bezier line to the path from last point to the point
+ * @param path              pointer to a path
+ * @param p1                pointer to a `lv_fpoint_t` variable for control point
+ * @param p2                pointer to a `lv_fpoint_t` variable for end point
+ */
+void lv_vector_path_quad_to(lv_vector_path_t * path, const lv_fpoint_t * p1, const lv_fpoint_t * p2);
+
+/**
+ * Add a cubic bezier line to the path from last point to the point
+ * @param path              pointer to a path
+ * @param p1                pointer to a `lv_fpoint_t` variable for first control point
+ * @param p2                pointer to a `lv_fpoint_t` variable for second control point
+ * @param p3                pointer to a `lv_fpoint_t` variable for end point
+ */
+void lv_vector_path_cubic_to(lv_vector_path_t * path, const lv_fpoint_t * p1, const lv_fpoint_t * p2,
+                             const lv_fpoint_t * p3);
+
+/**
+ * Close the sub path
+ * @param path              pointer to a path
+ */
+void lv_vector_path_close(lv_vector_path_t * path);
+
+/**
+ * Add a rectangle to the path
+ * @param path              pointer to a path
+ * @param rect              pointer to a `lv_area_t` variable
+ * @param rx                the horizontal radius for rounded rectangle
+ * @param ry                the vertical radius for rounded rectangle
+ */
+void lv_vector_path_append_rect(lv_vector_path_t * path, const lv_area_t * rect, int32_t rx, int32_t ry);
+
+/**
+ * Add a circle to the path
+ * @param path              pointer to a path
+ * @param c                 pointer to a `lv_fpoint_t` variable for center of the circle
+ * @param rx                the horizontal radius for circle
+ * @param ry                the vertical radius for circle
+ */
+void lv_vector_path_append_circle(lv_vector_path_t * path, const lv_fpoint_t * c, int32_t rx, int32_t ry);
+
+/**
+ * Add an sub path to the path
+ * @param path              pointer to a path
+ * @param subpath           pointer to another path which will be added
+ */
+void lv_vector_path_append_path(lv_vector_path_t * path, const lv_vector_path_t * subpath);
+
+/**
+ * Create a vector graphic descriptor
+ * @param layer         pointer to a layer
+ * @return              pointer to the created descriptor
+ */
+lv_vector_dsc_t * lv_vector_dsc_create(lv_layer_t * layer);
```

Comment:
The other draw related function has names like this: `lv_draw_rect_dsc_init`

So I wonder if we can
1. Use it as an `init` function: `lv_vector_dsc_init(&dsc)`
2. Set the layer in `lv_draw_vector(&dsc, layer)`. See [label draw](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_draw_label.h#L115-L116)
3. For consistency with other draw functions use `lv_draw_vector_dsc_*` instead of `lv_vector_dsc_*` (should be just a search and replace)

Reply:
I don't think that's a good idea.

1. The struct lv_vector_dsc_t is too complex, which contains data structures such as list Pointer and dynamic array，and is not suitable for _init initialization.

2. Passing in the layer when creating dsc is to get scissor_area for clipping, I don't think it's a bad idea to bind ahead of time.

3. lv_draw_vector as a prefix causes the function name to be too long. I think lv_vector is enough to distinguish it from other drawing functions. By the way, I still like lv_vector_draw not lv_draw_vector ：）

Reply:
I don't have a strong opinion on these. We should add these updates only if they doesn't cause too much trouble to keep up consistency.